activestorage/lib/active_storage/service/disk_service.rb

```diff
@@ -102,6 +102,16 @@ def path_for(key) #:nodoc:
       File.join root, folder_for(key), key
     end
 
+    def compose(*source_keys, destination_key)
+      File.open(make_path_for(destination_key), "w") do |file|
+        source_keys.each do |source_key|
+          download(source_key) do |chunk|
+            file << chunk
```

Comment:
If `download` guaranteed to be a local file here? Because if so you could use [`IO.copy_stream`](https://ruby-doc.org/core-2.7.0/IO.html#method-c-copy_stream) which would be way faster than having to read the files in user land.

Reply:
TIL. The method also seems to append when you're copying multiple streams, so it should work. 👍